agent/internal/config/jwt.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type JWTToken struct {
+	Header        JWTHeader
+	Payload       JWTPayload
+	Signature     string
+	InitialString string
+}
+
+type JWTHeader struct {
+	Alg string `json:"alg"`
+	Typ string `json:"typ,omitempty"`
+}
+
+type JWTPayload struct {
```

Comment:
If I recall correctly we do not use fields like this, as we are using JWTs for an initial auth (between agent and backend). I'll get back to you tomorrow.

Reply:
yes, the code doesn't utilize those, only `initialString` with the separate jwt library when parse with claims.

those types are made specifically for validation to ensure the JWT token has proper anatomy.

Reply:
[the codebase currently works with the string](https://github.com/dyrector-io/dyrectorio/pull/228/files#diff-e0c54bda26f4cc7eaa531f24a5237487e30bec13a10d76f23ef5662eb6c1b794R77) and [there is an external library JWT parsing](https://github.com/dyrector-io/dyrectorio/pull/228/files#diff-e0c54bda26f4cc7eaa531f24a5237487e30bec13a10d76f23ef5662eb6c1b794R64) so my types are not doing much except some explicitness and validation purposes.